### PR TITLE
Group tree 309

### PR DESCRIPTION
**Pull request by davidfraenkel**
## Description

Users need access to courses they’ve downloaded while offline or when they don’t want to use the internet. Currently, there is no option to store or access courses offline temporarily, limiting their ability to study when internet access is unavailable. Users also need a way to manage and delete downloaded content, with the option for automatic removal after a set period.

## Changes

Enable users to download and access courses offline. When offline, users should only see content they’ve downloaded. The app should store the content temporarily and delete it automatically after one month unless the user chooses to delete it manually.

### Acceptance critea
- [x] App switches to "Offline Mode" when offline, displaying only downloaded content.
- [x] "Downloads" page is accessible from the profile page, showing downloaded courses.
- [x] Download status and progress are visible.
- [x] Users can manually delete downloaded courses from the "Downloads" page.
- [x] Downloaded content auto-deletes after 30 days

https://github.com/user-attachments/assets/9d2ebc6a-f2e8-4eb5-b3e5-2239ffa70419

https://github.com/user-attachments/assets/91ef82c6-4ef0-48b4-b832-2be8aa167afe

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

is running on newest dev backend

Branch: Group-Tree-309

Commit id: 3b3fa4c

### CODE REVIEW

**Comment by Laskiri** [comment]
https://github.com/user-attachments/assets/a2b5417c-fa6f-4e11-82b9-029813bc465f
When signing into the app, subscribed courses no longer shows up automatically it just keeps on loading until you\ve been to one of the others screens and goes back.

File: eml/test.json
Comment (Laskiri) [inline eml/test.json:1]: Why is this added? what is it used for

Comment (davidfraenkel) [reply eml/test.json:1]: Sorrry, that was Lokes test file used for nothing so I deleted it :)

**Review by Laskiri — APPROVED** [review]
In general i think the added code looks fine , and if the current implementation has been validated by Iara then i would approve merging it into dev after adressing the "test.json" file. I am however concered about the fact that subscribed courses doesn't load on sign in anymore with these additions, and i think the reason to this needs to be found before moving forwards to staging.

**Comment by Laskiri** [comment]
> Ok, I found the fix for the bug but don't have the time to implement it another way to prevent the loading before the sprint meeting :) thank you for pointing out

No worries, I think you've done a great job though! 
Also i apologize for not reviewing the PR quicker, it kinda got lost in my mail with all the other PR's i've been checking today 😅 Next time, if you write me a short dm on discord or messenger,  I promise to get back to it in a flash ⚡

**Review by Laskiri — APPROVED** [review]
Nice work, looks great :D
